src/main/java/org/javamodularity/moduleplugin/tasks/CompileJavaTaskMutator.java

```diff
@@ -44,6 +50,25 @@ void modularizeJavaCompileTask(JavaCompile javaCompile) {
         // https://github.com/java9-modularity/gradle-modules-plugin/issues/45
         helper().findTask(COMPILE_KOTLIN_TASK_NAME, AbstractCompile.class)
                 .ifPresent(compileKotlin -> javaCompile.setDestinationDir(compileKotlin.getDestinationDir()));
+
+        configureGroovy(javaCompile, compilerArgs);
+    }
+
+    private void configureGroovy(JavaCompile javaCompile, List<String> compilerArgs) {
+        GroovyCompile compileGroovy = (GroovyCompile)project.getTasks().findByName(COMPILE_GROOVY_TASK_NAME);
+        if (compileGroovy != null) {
+            JavaPluginConvention javaConvention = project.getConvention().getPlugin(JavaPluginConvention.class);
+            for(String sourceSetName: List.of(SourceSet.MAIN_SOURCE_SET_NAME, SourceSet.TEST_SOURCE_SET_NAME)) {
+                SourceSet sourceSet = javaConvention.getSourceSets().getByName(sourceSetName);
+                sourceSet.getJava().include("module-info.java");
+                javaCompile.setDestinationDir(compileGroovy.getDestinationDir());
+
+                GroovySourceSet groovySourceSet = (GroovySourceSet)new DslObject(sourceSet).getConvention().getPlugins().get("groovy");
+                groovySourceSet.getGroovy().setSrcDirs(List.of("src/" + sourceSetName + "/java", "src/" + sourceSetName + "/groovy"));
```

Comment:
Would you mind explaining why you need to modify the source dirs of the main and test Groovy source sets?

I'm wondering because:
- it was solved differently for Kotlin
- I'm afraid it may break my project, where I always apply all three plugins:
   1. Java (for main code)
   2. Groovy (for Spock)
   3. Kotlin (for test helpers)

Right now, I'm [testing on classpath only](https://github.com/tlinkowski/tlinkowski-superpom/blob/4a692ac44ba0a78dd838aacdb7055e50112f6dbf/subprojects/my-superpom-gradle-plugin/src/main/kotlin/pl/tlinkowski/gradle/my/superpom/shared/internal/plugin/ModularityConfigPlugin.kt#L52-L58), because Groovy 2 doesn't work on modulepath.

I specifically [configured `compileTestGroovy` to have the results of `compileTestKotlin` on its classpath](https://github.com/tlinkowski/tlinkowski-superpom/blob/4a692ac44ba0a78dd838aacdb7055e50112f6dbf/subprojects/my-superpom-gradle-plugin/src/main/kotlin/pl/tlinkowski/gradle/my/superpom/shared/internal/plugin/TestConfigPlugin.kt#L112-L125). I wonder whether such approach would work on modulepath (but that's my problem to figure it out - here, I'd only like to make sure that this plugin doesn't make any incompatible assumptions).